### PR TITLE
Add rust-side logger

### PR DESCRIPTION
**Pull request by DanieleDiBenedetto**
Added log4rs support and a new `init_zendoo()` function to be called at node startup. The function is generic but currently it just initializes a logger with a configuration using a .yaml file supposed to be supplied by the caller.
Then, I basically replaced the `print!` and `println!` with `log::debug!` or `log::info!` (according to the situation) and the `eprint!` and `eprintln!` with `log::error!`

### CODE REVIEW

File: src/tests/log/sample_log_config.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  file:
+    kind: rolling_file
+    path: test_log.log
+    encoder:
+      pattern: "{d(%Y-%m-%d %H:%M:%S)} | {({l}):5.5} | {f}:{L} — {m}{n}"
+    policy:
+      trigger:
+        kind: size
+        limit: 10 kb
```
Comment (la10736) [inline src/tests/log/sample_log_config.yaml:11]: Are you sure that is a right default?

**Review by la10736 — CHANGES_REQUESTED** [review]
Also in this case you must decide if you want to use log in tests or printf.

One more thigh... you should pay more attention to not mix formatting and code changes.

File: src/tests/log/sample_log_config.yaml (hunk 1 above)
Comment (DanieleDiBenedetto) [reply src/tests/log/sample_log_config.yaml:11]: Just used to test correct generation of multiple log files

**Comment by la10736** [comment]
Build fail on travis... maybe you forget to include the new version that implements `num_proofs()`

**Review by ptagl — APPROVED** [review]
It looks good to me, thanks!

**Review by la10736 — APPROVED** [review]

**Pull request merged**